src/Factory/ShopBillingDataFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Factory;
+
+use Sylius\Component\Resource\Exception\UnsupportedMethodException;
+use Sylius\RefundPlugin\Entity\ShopBillingData;
+use Sylius\RefundPlugin\Entity\ShopBillingDataInterface;
+
+final class ShopBillingDataFactory implements ShopBillingDataFactoryInterface
+{
+    public function createNew(): ShopBillingDataInterface
+    {
+        throw new UnsupportedMethodException('This object is not default constructable.');
+    }
+
+    public function createWithData(
+        ?string $company,
+        ?string $taxId,
+        ?string $countryCode,
+        ?string $street,
+        ?string $city,
+        ?string $postcode
+    ): ShopBillingDataInterface {
+        $shopBillingData = new ShopBillingData();
```

Comment:
You should use here `className` similar to: https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Order/Factory/OrderItemUnitFactory.php#L34 or go back to decorating the factory

Reply:
As `$className` is private property (😞) of a parent class, we should probably override a constructor and assign it to the property here as well. But yes, using this parameter is required 👌

Reply:
its decorated now